Automattic/Co-Authors-Plus
# only update authors on post save

## Description

Based on https://github.com/eupolitico/Co-Authors-Plus/commit/230b1f354b22f41a771e4751c1a507fcd1e251c9

This fixes a regression added in https://github.com/Automattic/Co-Authors-Plus/pull/870 that would immediately update the authors of a post once you selected them in the sidebar.

With this PR, authors will only be updated when saving the post.

Note: This does not fix a bug that was present that will trigger the author update also for auto-saves.

## Deploy Notes

Nothing to note.

## Steps to Test

1. Edit an existing post
2. Make changes to the author list (add or remove an author)
3. Confirm that the change is not applied until you save the post

cc @elliott-stocks @jomurgel
